includes/class-wc-stripe-payment-method-configurations.php

```diff
@@ -362,16 +356,24 @@ public static function maybe_migrate_payment_methods_from_db_to_pmc( $force_migr
 			);
 		}
 
-		// Update the PMC if there are any enabled payment methods
+		// Update the PMC if there are locally enabled payment methods
 		if ( ! empty( $enabled_payment_methods ) ) {
-
 			// Get all available payment method IDs from the configuration.
 			// We explicitly disable all payment methods that are not in the enabled_payment_methods array
 			$available_payment_method_ids = [];
 			foreach ( $merchant_payment_method_configuration as $payment_method_id => $payment_method ) {
 				if ( isset( $payment_method->display_preference ) ) {
 					$available_payment_method_ids[] = $payment_method_id;
 				}
+
+				// We want to also include payment methods enabled in the PMC, except for express payment methods.
+				if (
+					! in_array( $payment_method_id, WC_Stripe_Payment_Methods::EXPRESS_PAYMENT_METHODS, true ) &&
+					! in_array( $payment_method_id, $enabled_payment_methods, true ) &&
+					isset( $payment_method->display_preference->value ) && 'on' === $payment_method->display_preference->value
+				) {
+					$enabled_payment_methods[] = $payment_method_id;
+				}
 			}
 
 			self::update_payment_method_configuration(
```

Comment:
Not blocking: might it be worth logging an error to note what the merged configuration looks like? (I considered a `warning`, but it looks like that is gated behind the logging flag, so would be less useful.)
```suggestion
			WC_Stripe_Logger::error( 'Switching to Stripe-hosted payment method configuration', [ 'pmc_id' => $merchant_payment_method_configuration->id, 'enabled_payment_methods' => $enabled_payment_methods ] );
			self::update_payment_method_configuration(
```

Reply:
We should also log `available_payment_method_ids`:

```suggestion
			WC_Stripe_Logger::error(
				'Switching to Stripe-hosted payment method configuration',
				[
					'pmc_id'                       => $merchant_payment_method_configuration->id,
					'enabled_payment_methods'      => $enabled_payment_methods,
					'available_payment_method_ids' => $available_payment_method_ids,
				]
			);
			self::update_payment_method_configuration(
```